### PR TITLE
Depend re-structure

### LINKED ISSUE #79: Refactor: Create `parser` a subset of `resolver`
In the current architecture, `depend.core` first triggers `depend.parser` to parse dependency versions, and then uses `depend.resolver` to fetch exact versions and their subdependencies.

But due to this structure, `resolver` also has to use `parser`, to parse the API responses and so on.

Because of this, it's better for `resolver` to be the top level module, and `parser` only be used internally by it.

### PR DESCRIPTION
**Pull request by NobleMathews**
Fixes #79

### CODE REVIEW

File: src/depend/cli.py [hunk 1]
```diff
@@ -55,13 +55,16 @@ def main(
 
     """
     try:
-        if dep_file:
-            result = Depend.from_dependency_file(
```
Comment (tushar-deepsource) [inline src/depend/cli.py:59]: Don't remove this alternate constructor, this was fine.

Comment (tushar-deepsource) [reply src/depend/cli.py:59]:
```python
@classmethod
def from_dep_file(cls, file, *args, **kwargs):
    dep_content = handle_dep_file(file)
    packages = dep_content.get("pkg_dep")
    file_extension = os.path.basename(file).split(".")[-1]
    if file_extension == "lock" or "lock" in os.path.basename(file):
        cls(packages, *args, depth=1, **kwargs)
    return cls(packages, *args, **kwargs)
```
You can define that method like this.

Comment (tushar-deepsource) [reply src/depend/cli.py:59]: With the comments that already exist in core.py.

Comment (NobleMathews) [reply src/depend/cli.py:59]: Isn't it more straightforward to keep the single constructor though?  also the reason i removed it was mostly to not call handle_dep_file within the cli that comes from the parser

Comment (tusharsadhwani) [reply src/depend/cli.py:59]: Having two constructors is better than having one constructor with the required arguments also being optional (both `packages` and `dep_file` have a default `None`)

You'll throw an exception when none are given, and ignore one of them when both are given. Two constructors are much easier that way

Comment (tushar-deepsource) [reply src/depend/cli.py:59]: > to not call `handle_dep_file` within the cli (that comes from the parser)
For now you can create a `read_dep_file` or some such function inside resolver and call that instead. We'll improve it further later.

**Comment by tushar-deepsource** [comment]
Closing and opening to trigger tests.

File: src/depend/resolver/__init__.py [hunk 2]
```diff
@@ -282,9 +282,13 @@ async def work_with_vers(
         return result_list, rem_dep
 
 
+def resolve_dep_file(file):
```
Comment (tushar-deepsource) [inline src/depend/resolver/__init__.py:285]:
```suggestion
def parse_dep_file(file):
```

File: src/depend/__init__.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import enum
+import os
+import pathlib
+import sys
+from typing import Optional
+
+from loguru import logger
+
+from depend.errors import FileNotSupportedError, LanguageNotSupportedError
+from depend.resolver import make_multiple_requests, resolve_dep_file
```
Comment (tushar-deepsource) [inline src/depend/__init__.py:13]:
```suggestion
from depend.resolver import make_multiple_requests, parse_dep_file
```

File: src/depend/__init__.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        }
+        logger.configure(**config)
+
+    @classmethod
+    def from_dep_file(cls, file: pathlib.Path, *args, **kwargs):
+        if not os.path.isfile(file):
+            raise FileNotSupportedError(
+                "Dependency information missing, please provide a valid dep_file or packages"
+            )
+        dep_content = resolve_dep_file(file)
```
Comment (tushar-deepsource) [inline src/depend/__init__.py:76]:
```suggestion
        dep_content = parse_dep_file(file)
```

**Pull request merged**